### PR TITLE
refactor(all): update repo structure

### PR DESCRIPTION
**Pull request by martingust**
Initial update of repo structure and build to align with other repos. Needs more work though: specifically on getting the tests to run and fix lint errors.

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
@martingust Should I give you push access to this repo?

**Comment by martingust** [comment]
@EisenbergEffect That would be great, thanks!

**Comment by MeirionHughes** [comment]
server team with push access? :dancer:

File: package.json [hunk 1]
```diff
@@ -1,34 +1,74 @@
 {
   "name": "aurelia-pal-nodejs",
-  "description": "node-js implementation of the aurelia pal-abstraction layer",
-  "author": "Meirion Hughes <crakinshot@yahoo.com>",
+  "version": "0.1.0",
```
Comment (MeirionHughes) [inline package.json]: 1.0.0-alpha.1 is already live (and used) on npm unfortunatly.

**Comment by EisenbergEffect** [comment]
@martingust @MeirionHughes The server team is setup and you are both added. I'll let you two handle this PR.

**Comment by devanp92** [comment]
@martingust @MeirionHughes I implemented _cycleTidy + MutationObserver.takeRecords() based off @martingust's ```structure``` branch. I can make a PR when this is merged (to avoid conflicts).

File: package.json (hunk 1 above)
Comment (martingust) [inline package.json]: Ok, should we just keep use the `alpha` tag and release a `1.0.0-alpha.2.0.0` next? @MeirionHughes @EisenbergEffect

**Comment by martingust** [comment]
@MeirionHughes @devanp92 This is merged.

**Comment by MeirionHughes** [comment]
okay cool; @devanp92 I think the best option for the mutation would be to polyfill it. As I mentioned,  I tried and failed to get this working ontop of jsdom's global/window...  https://github.com/Polymer/MutationObservers

**Comment by devanp92** [comment]
@MeirionHughes Do you think that this might be a better approach:

Something like

```
import {jsdom} from 'jsdom';
import {fs} from 'fs';
...

const mutationObserver = fs.readFileSync('./mutationobserver.js'); //Polymer Polyfill
jsdom.env({
    src: [mutationObserver],
    done: (err, window) => {
        _global = window;
    }
});
``` 

I believe that  ```window``` would contain MutationObserver since Polymer's polyfill binds the MutationObserver on the global object.

In [the example on jsdom's README](https://github.com/tmpvar/jsdom#easymode-jsdomenv), they pass in jQuery as an external resource to be used with jsDom.

**Comment by MeirionHughes** [comment]
ah ok cool; I'll quickly try.

**Comment by MeirionHughes** [comment]
@martingust lost intellisense now that all the files are js rather than js. :(

**Comment by martingust** [comment]
@MeirionHughes 

- js rather than ts? That is to be align with all other repos that are only js. Any idea what needed to be done to get back intellisense in vscode for this repo?

- No compile errors at all? Here is what I'm getting: 
screen shot 2016-08-28 at 20 14 28

- This was also to be align with the other repos. But CommonJs might be all that's needed?

**Pull request merged**